### PR TITLE
Support dumping databases with std::duration config vals

### LINKED ISSUE #5520: Dump fails with `error receiving dump header: unexpected constant type: <class 'edb.ir.statypes.Duration'>`
- EdgeDB Version: `2.14` and `2.15`
- EdgeDB CLI Version: `2.3.1+5d93f42` and `3.0.0+80bc8b4`
- OS Version: macos 13

Steps to Reproduce:

Unsure as to the exact circumstances that led to this so apologies in advance for lack of detail. This does work as expected on an instance with identical version/schema, so this may be data dependent.

1. Run `edgedb dump -I instancename instance.dump`

```
edgedb error: InternalServerError: error receiving dump header: unexpected constant type: <class 'edb.ir.statypes.Duration'>
  Hint: This is most likely a bug in EdgeDB. Please consider opening an issue ticket at https://github.com/edgedb/edgedb/issues/new?template=bug_report.md
  Server traceback:
      Traceback (most recent call last):
        File "/usr/lib/x86_64-linux-gnu/edgedb-server-2/lib/python3.10/site-packages/edb/server/compiler_pool/worker_proc.py", line 55, in worker
          res = meth(*args)
        File "/usr/lib/x86_64-linux-gnu/edgedb-server-2/lib/python3.10/site-packages/edb/server/compiler/compiler.py", line 2595, in describe_database_dump
          config_ddl = config.to_edgeql(config.get_settings(), database_config)
        File "/usr/lib/x86_64-linux-gnu/edgedb-server-2/lib/python3.10/site-packages/edb/server/config/ops.py", line 394, in to_edgeql
          val = value_to_edgeql_const(setting, value.value)
        File "/usr/lib/x86_64-linux-gnu/edgedb-server-2/lib/python3.10/site-packages/edb/server/config/ops.py", line 333, in value_to_edgeql_const
          ql = s_utils.const_ast_from_python(value)
        File "/usr/lib/x86_64-linux-gnu/edgedb-server-2/lib/python3.10/site-packages/edb/schema/utils.py", line 1128, in const_ast_from_python
          raise ValueError(f'unexpected constant type: {type(val)!r}')
      ValueError: unexpected constant type: <class 'edb.ir.statypes.Duration'>
```

Schema:

Can send schema by discord DM Jack Fischer#5469. Happy to pull any other info that would be helpful. No `duration`s in schema.

### PR DESCRIPTION
**Pull request by msullivan**
Fixes #5520.

### CODE REVIEW

File: edb/testbase/server.py [hunk 1]
```diff
@@ -1434,6 +1434,8 @@ async def check_dump_restore(self, check_method):
                 self.run_cli, '-d', src_dbname, 'dump', f.name
             )
 
+            # breakpoint()
```
Comment (elprans) [inline edb/testbase/server.py:1437]:
```suggestion
```

**Comment by msullivan** [comment]
@fantix The hack I described as managing to fix the weird state syncing issue (executing a query before entering the transaction) only seems to work when I'm running the test alone. (Which is to say, doesn't really work at all.)

I tried inspecting `cfg::DatabaseConfig.query_execution_timeout` (instead of `cfg::DatabaseConfig.query_execution_timeout`), and that gives an empty set. 
This is pretty weird on its own since `query_execution_timeout` is marked as `required` (though it doesn't have a default).

Could you take a look?

**Comment by fantix** [comment]
I can reproduce locally, and I feel like there are more than 1 issue. I'm still debugging.

**Comment by fantix** [comment]
There's another issue with the dump/restore test infra itself, I'm debugging to push another fix.

**Review by fantix — APPROVED** [review]
The failing test should be fixed now.

**Review by elprans — APPROVED** [review]

**Comment by msullivan** [comment]
I confirmed it doesn't seem to affect test time

**Pull request merged**